src/main/java/spoon/support/compiler/jdt/PositionBuilder.java

```diff
@@ -133,19 +139,16 @@ SourcePosition buildPositionCtElement(CtElement e, ASTNode node) {
 			int bodyStart = typeDeclaration.bodyStart;
 			int bodyEnd = typeDeclaration.bodyEnd;
 
-			Annotation[] annotations = typeDeclaration.annotations;
-			if (annotations != null && annotations.length > 0) {
-				if (annotations[0].sourceStart() == declarationSourceStart) {
-					modifiersSourceStart = findNextNonWhitespace(contents, declarationSourceEnd, annotations[annotations.length - 1].declarationSourceEnd + 1);
-				}
-			}
-			if (modifiersSourceStart == 0) {
+			if (modifiersSourceStart <= 0) {
 				modifiersSourceStart = declarationSourceStart;
 			}
 			//look for start of first keyword before the type keyword e.g. "class". `sourceStart` points at first char of type name
 			int modifiersSourceEnd = findPrevNonWhitespace(contents, modifiersSourceStart - 1,
 										findPrevWhitespace(contents, modifiersSourceStart - 1,
 											findPrevNonWhitespace(contents, modifiersSourceStart - 1, sourceStart - 1)));
+			if (e instanceof CtModifiable) {
+				setModifiersPosition((CtModifiable) e, modifiersSourceStart, bodyStart);
```

Comment:
I have not read whole code yet, so may be it is correct, but usage of `bodyStart` here seems to be wrong.

Reply:
ok, the searched string is little bit longer then necessary, but I cannot imagine code where it would cause problem, so it is OK for me.